### PR TITLE
Fix failed tests related to google_service_networking_connection

### PR DESCRIPTION
**Pull request by zli82016**
The previous PR https://github.com/GoogleCloudPlatform/magic-modules/pull/8872 caused some nightly tests failed. 

1. To delete the `google_service_networking_connection` resource,  call the method deleteConnection. If it fails, return the error and the deletion fails. Based on the test results, only sql instance has a waiting period to delete the connection.
2. Use bootstrapped service networking connection for handwritten tests. 
3. For mm1 generated tests, create a new service networking connection, which can be deleted with deleteConnection. The reason to not use bootstrapped service networking connection is to not modify the examples in the doc. 

5.0 guide is in https://github.com/GoogleCloudPlatform/magic-modules/pull/9009

**Release Note Template for Downstream PRs (will be copied)**

```release-note:breaking-change
servicenetworking: used the `deleteConnection` method to delete the resource `google_service_networking_connection`
```

### CODE REVIEW

**Review by shuyama1 — COMMENTED** [review]
As discussed offline, I believe bootstrapping is still needed for this PR

File: mmv1/third_party/terraform/services/servicenetworking/resource_service_networking_connection.go [hunk 1]
```diff
@@ -261,42 +261,69 @@ func resourceServiceNetworkingConnectionDelete(d *schema.ResourceData, meta inte
...
-	res, err := transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
-		Config:    config,
-		Method:    "POST",
-		Project:   project,
-		RawURL:    url,
-		UserAgent: userAgent,
-		Body:      obj,
-		Timeout:   d.Timeout(schema.TimeoutDelete),
-	})
+	connectionId, err := parseConnectionId(d.Id())
```
[the remark this answers is not in the document]
Comment (zli82016) [reply mmv1/third_party/terraform/services/servicenetworking/resource_service_networking_connection.go:270]: Right. Previously, `removePeering` for deletion of the this resource.
Now deleteConnection is called first. If it failed, `removePeering` is called. 

I don't know if there are any cases that `removePeering` should not be called.

[the remark this answers is not in the document]
Comment (zli82016) [reply mmv1/third_party/terraform/services/servicenetworking/resource_service_networking_connection.go:268]: I will catch the error when `deleteConnection` failed and not call `removePeering`

**Comment by zli82016** [comment]
I am going to fix the test `TestAccVertexAIIndexEndpoint_updated`  in a separate PR. It is related to labels.

**Comment by zli82016** [comment]
5.0 guide is in https://github.com/GoogleCloudPlatform/magic-modules/pull/9009

**Comment by zli82016** [comment]
The failed test is unrelated.

File: mmv1/third_party/terraform/services/servicenetworking/resource_service_networking_connection.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (zli82016) [reply mmv1/third_party/terraform/services/servicenetworking/resource_service_networking_connection.go:268]: I am going to return the error and the deletion will fail.

**Comment by zli82016** [comment]
The failed tests are unrelated.

File: mmv1/third_party/terraform/services/alloydb/resource_alloydb_backup_test.go [hunk 2]
```diff
@@ -53,41 +52,58 @@ resource "google_alloydb_backup" "default" {
...
+}
+
+data "google_compute_network" "default" {
+  name = "%{network_name}"
+}
+`, context)
+}
+
+// Updates "label" field
+func testAccAlloydbBackup_update(context map[string]interface{}) string {
```
Comment (shuyama1) [inline mmv1/third_party/terraform/services/alloydb/resource_alloydb_backup_test.go:79]: is this updating labels change intended here?

**Review by shuyama1 — APPROVED** [review]

File: mmv1/third_party/terraform/services/alloydb/resource_alloydb_backup_test.go (hunk 2 above)
Comment (zli82016) [reply mmv1/third_party/terraform/services/alloydb/resource_alloydb_backup_test.go:79]: Right, the labels updating test exists previously.

**Pull request merged**